portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java

```diff
@@ -1158,6 +1173,13 @@ public DLFolder updateStatus(
 
 		dlFolderPersistence.update(dlFolder);
 
+		// Parent folder
+
+		if (status == WorkflowConstants.STATUS_APPROVED) {
```

Comment:
We need to check that `dlFolder.getParentFolderId()` is not 0 because otherwise this will fail?

Reply:
You're right, I will add the other condition to the if statement.